cmd/cerbos/compile/internal/verification/display.go

```diff
@@ -0,0 +1,184 @@
+					p.Printf("==== %s ", colored.Action(action.Name))
+					if action.Data.Skipped {
+						p.Println(colored.SkippedTest("[SKIPPED]"))
+						continue
+					}
+
+					if action.Data.Failed {
+						p.Println(colored.FailedTest("[FAILED]"))
+						p.Printf("\tError: %s\n", action.Data.Error)
+						if verbose {
+							p.Printf("\tTrace: \n%s\n", action.Data.EngineTrace)
+						}
+						continue
+					}
+
+					p.Println(colored.SuccessfulTest("[OK]"))
+				}
+			}
+		}
+
+		if sn.Failed {
+			p.Println(colored.ErrorMsg("Invalid test suite"))
+		}
+	}
+
+	if result.Failed {
+		return errors.ErrTestsFailed
+	}
+
+	return nil
+}
+
+func displayTree(p *printer.Printer, result *verify.Result, verbose bool) error {
+	tree := pterm.LeveledList{}
+
+	p.Println(colored.Header("Test results"))
+	for _, sn := range result.Suites {
+		suiteText := fmt.Sprintf("%s (%s)", colored.Suite(sn.Name), colored.FileName(sn.File))
+		if sn.Failed {
+			suiteText = fmt.Sprintf("%s %s", suiteText, colored.FailedTest("[FAILED]"))
+			tree = append(tree, pterm.LeveledListItem{
+				Level: 0,
+				Text:  suiteText,
+			})
+			tree = append(tree, pterm.LeveledListItem{
+				Level: 1,
+				Text:  colored.ErrorMsg(sn.Status),
+			})
+			continue
+		}
+
+		if sn.Skipped {
+			suiteText = fmt.Sprintf("%s %s", suiteText, colored.SkippedTest("[SKIPPED]"))
+			tree = append(tree, pterm.LeveledListItem{
+				Level: 0,
+				Text:  suiteText,
+			})
+			continue
+		}
+
+		tree = append(tree, pterm.LeveledListItem{
+			Level: 0,
+			Text:  suiteText,
+		})
+
+		for _, principal := range sn.Principals {
+			tree = append(tree, pterm.LeveledListItem{
+				Level: 1,
+				Text:  colored.Principal(principal.Name),
+			})
+			for _, resource := range principal.Resources {
+				tree = append(tree, pterm.LeveledListItem{
+					Level: 2, //nolint:gomnd
+					Text:  colored.Resource(resource.Name),
+				})
+				for _, action := range resource.Actions {
+					actionText := colored.Action(action.Name)
+
+					if action.Data.Failed {
+						actionText = fmt.Sprintf("%s %s", actionText, colored.FailedTest("[FAILED]"))
+					}
+
+					if action.Data.Skipped {
+						actionText = fmt.Sprintf("%s %s", actionText, colored.SkippedTest("[SKIPPED]"))
+					}
+
+					if !action.Data.Failed && !action.Data.Skipped {
+						actionText = fmt.Sprintf("%s %s", actionText, colored.SuccessfulTest("[OK]"))
+					}
+
+					tree = append(tree, pterm.LeveledListItem{
+						Level: 3, //nolint:gomnd
+						Text:  actionText,
+					})
+
+					if action.Data.Failed {
+						tree = append(tree, pterm.LeveledListItem{
+							Level: 4, //nolint:gomnd
+							Text:  fmt.Sprintf("%s %s", colored.ErrorMsg("ERROR:"), action.Data.Error),
+						})
+						if verbose && action.Data.EngineTrace != "" {
+							tree = append(tree, pterm.LeveledListItem{
+								Level: 4, //nolint:gomnd
+								Text:  fmt.Sprintf("%s %s", colored.Trace("TRACE:"), action.Data.EngineTrace),
+							})
+						}
```

Comment:
The tree kind of breaks down when there are traces to display. We should either print the traces at the end as a separate list OR format the traces as tree nodes so that they display nicely. I think the first option might be easier to implement and read.

```
Test results
└─┬AlbumObjectTestSuite (album_object_test.yaml)
  ├─┬alicia
  │ ├─┬alicia_private_album
  │ │ ├─┬view [FAILED]
  │ │ │ ├──ERROR:   string(
- 	"EFFECT_DENY",
+ 	"EFFECT_ALLOW",
  )

  │ │ │ └──TRACE: policy=cerbos.resource.album_object.vdefault > scope="" > derivedRole=owner > condition > expr=`request.resource.attr.owner == request.principal.id`
	result -> true

policy=cerbos.resource.album_object.vdefault > scope="" > derivedRole=owner
	activated -> true
```